internal/workflow.go

```diff
@@ -472,6 +472,36 @@ type (
 		VersioningBehavior VersioningBehavior
 	}
 
+	// LoadDynamicOptionsDetails is used as input to the LoadDynamicOptions callback for dynamic workflows
+	//
+	// Exposed as: [go.temporal.io/sdk/workflow.LoadDynamicOptionsDetails]
+	LoadDynamicOptionsDetails struct {
+		WorkflowType WorkflowType
+	}
+
+	// DynamicRegisterOptions consists of options for registering a dynamic workflow
+	//
+	// Exposed as: [go.temporal.io/sdk/workflow.DynamicRegisterOptions]
+	DynamicRegisterWorkflowOptions struct {
+		// Allows dynamic options to be loaded for a workflow.
+		LoadDynamicOptions func(details LoadDynamicOptionsDetails) (RegisterDynamicWorkflowOptions, error)
```

Comment:
The more I think about this, the concept of `DynamicRegisterWorkflowOptions` (even if exposed as `DynamicRegisterOptions`) and `RegisterDynamicWorkflowOptions` is just too confusingly named.

I think we should use the word "runtime" in here maybe, e.g.:

```suggestion
		LoadDynamicRuntimeOptions func(details LoadDynamicRuntimeOptionsDetails) (DynamicRuntimeWorkflowOptions, error)
```

Reply:
Fine with me, whatever we do here we should also change the dynamic activity options name

Reply:
This is only specifically concerning runtime loading of workflow options by workflow type and has no activity anallog at this time